components/Navigation.js

```diff
@@ -0,0 +1,233 @@
+import React from "react";
+
+import Link from "next/link";
+import Drawer from "@material-ui/core/Drawer";
+import Button from "@material-ui/core/Button";
+import List from "@material-ui/core/List";
+import ListItem from "@material-ui/core/ListItem";
+import ListItemText from "@material-ui/core/ListItemText";
+import Hidden from "@material-ui/core/Hidden";
+import Toolbar from "@material-ui/core/Toolbar";
+import useMediaQuery from "@material-ui/core/useMediaQuery";
+import IconButton from "@material-ui/core/IconButton";
+import Menu from "@material-ui/core/Menu";
+import MenuItem from "@material-ui/core/MenuItem";
+import MenuList from "@material-ui/core/MenuList";
+import AppBar from "@material-ui/core/AppBar";
+import Collapse from "@material-ui/core/Collapse";
+
+import ExpandMore from "@material-ui/icons/ExpandMore";
+import ExpandLess from "@material-ui/icons/ExpandLess";
+
+import MenuIcon from "@material-ui/icons/Menu";
+
+import { makeStyles, useTheme } from "@material-ui/core/styles";
+
+const drawerWidth = 190;
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    flexGrow: 1,
+    [theme.breakpoints.down("sm")]: {
+      width: drawerWidth,
+    },
+  },
+  appBar: {
+    backgroundColor: "white",
+    display: "flex",
+    flexWrap: "wrap",
+    overflowX: "auto",
+  },
+  menuButton: {
+    color: "#007cc4",
+    backgroundColor: "white",
+    margin: theme.spacing(2),
+    width: "100px",
+  },
+  navbarLogo: {
+    width: "200px",
+    margin: theme.spacing(2),
+  },
+  submenuItem: {
+    color: "#007cc4",
+  },
+  nested: {
+    paddingLeft: theme.spacing(4),
+  },
+}));
+
+export default function Navigation() {
+  const classes = useStyles();
+  const theme = useTheme();
+
+  const mdUp = useMediaQuery(theme.breakpoints.up("md"));
+
+  const [mobileOpen, setMobileOpen] = React.useState(false);
+  const [anchorEl, setAnchorEl] = React.useState(null);
+  const [dropdownOpen, setDropDownOpen] = React.useState(false);
+
+  const open = !!anchorEl;
+
+  const handleDrawerToggle = () => {
+    setMobileOpen(!mobileOpen);
+  };
+
+  const handleClick = (e) => {
+    setAnchorEl(e.currentTarget);
+  };
+
+  const handleDropdownClose = () => {
+    setAnchorEl(null);
+  };
+
+  const handleDropdownOpen = () => {
+    setDropDownOpen(!dropdownOpen);
+  };
+
+  const navigationItems = [
```

Comment:
This can be moved out of the function's definition (just above it in this file). Otherwise this large array if going to get recreated on every render. Otherwise you can write it in [useMemo](https://reactjs.org/docs/hooks-reference.html#usememo).